File: common/src/main/java/cc/turtl/chiselmon/feature/pc/tooltip/TooltipBuilder.java
package cc.turtl.chiselmon.feature.pc.tooltip;

import cc.turtl.chiselmon.config.category.PCConfig;
import cc.turtl.chiselmon.util.format.PokemonFormats;
import com.cobblemon.mod.common.pokemon.Pokemon;
import net.minecraft.client.gui.components.Tooltip;
import net.minecraft.network.chat.Component;
import net.minecraft.network.chat.MutableComponent;
import org.jetbrains.annotations.NotNull;

public final class TooltipBuilder {
    private TooltipBuilder() {
    }

    public static Tooltip build(@NotNull Pokemon pokemon, @NotNull PCConfig.TooltipConfig config, boolean shiftDown) {

        MutableComponent content = PokemonFormats.detailedName(pokemon).copy();

        if (shiftDown) {
            for (TooltipEntry entry : TooltipRegistry.getEntries()) {
                if (entry.shouldDisplay(config, pokemon)) {
                    content.append(Component.literal("\n")).append(entry.getComponent(pokemon));
                }
            }
        }

        return Tooltip.create(content);
    }
}

File: common/src/main/kotlin/cc/turtl/chiselmon/client/feature/pc/tooltip/TooltipBuilder.kt
package cc.turtl.chiselmon.client.feature.pc.tooltip

import cc.turtl.chiselmon.client.config.category.PCConfig
import cc.turtl.chiselmon.core.util.format.PokemonFormats.detailedName
import com.cobblemon.mod.common.pokemon.Pokemon
import net.minecraft.client.gui.components.Tooltip
import net.minecraft.network.chat.Component

object TooltipBuilder {
    @JvmStatic
    fun build(pokemon: Pokemon, config: PCConfig.TooltipConfig, shiftDown: Boolean): Tooltip {
        val content = detailedName(pokemon, false).copy()

        if (shiftDown) {
            TooltipRegistry.getEntries()
                .filter { it.shouldDisplay(config, pokemon) }
                .forEach { content.append(Component.literal("\n")).append(it.getComponent(pokemon)) }
        }

        return Tooltip.create(content)
    }
}